src/content/docs/en/guides/assets.mdx

```diff
@@ -0,0 +1,291 @@
+
+{/* astro
+---
+// src/pages/index.astro
+import { Image } from 'astro:assets';
+
+// Access images in `src/images/`
+import logo from '../assets/images/logo.png';
+---
+<Image src={logo} alt="Astro" /> */}
+
+
+### Update existing `<img>` tags
+
+Previously, importing an image would return a simple `string` with the path of the image. With the new `image` features enabled, imported image assets now match the following signature:
+
+```ts
+interface ImageAsset {
+  src: string;
+  width: number;
+  height: number;
+  format: string;
+}
+```
+
+You must update the `src` attribute of any existing `<img>` tags, and you may also update other attributes that are now available to you from the imported image.
+
+```astro title="src/components/MyComponent.astro" ".src" ".width" ".height" del={2,5} ins={3,6}
+---
+import rocket from '../images/rocket.svg';
+import rocket from '../assets/images/rocket.svg'
+---
+<img src={rocket} width="250" height="250" alt="A rocketship in space." />
+<img src={rocket.src} width={rocket.width} height={rocket.height} alt="A rocketship in space." />
+```
+
+### Update your Markdown files
+
+Relative images located in `src/assets` can now be referenced in Markdown and MDX files. These will automatically be optimized and hashed by Astro's build process.
+
+This allows you to move your images from the `public/` directory to the new, reserved `src/assets/` directory. (See the URL path of these built images in the example below.)
+
+Your existing images in `public/` and remote images are still valid, but are not automatically optimized by Astro's build process.
+
+```md title="src/pages/posts/post-1.md" "/_astro" ".hash" "../../assets/"
+# My Markdown Page
+
+<!-- Local image stored at src/assets/stars.png -->
+![A starry night sky.](../../assets/stars.png)
+<img src="/_astro/stars.hash.png" alt="A starry night sky.">
+
+<!-- Image stored at public/images/stars.png -->
+![A starry night sky.](/images/stars.png)
+<img src="/images/stars.png" alt="A starry night sky.">
+
+<!-- Remote image on another server -->
+![Astro logo](https://astro.build/assets/logo.png)
+<img src="https://astro.build/assets/logo.png" width="25" alt="Astro logo">
+```
+
+### Convert from `@astrojs/image`
+
+**Built-in asset support is incompatible with the `@astrojs/image` integration.**
+
+To avoid errors in your project, complete the following steps:
+
+1. Remove the `@astrojs/image` integration.
+
+    You must [remove the integration](/en/guides/integrations-guide/#removing-an-integration) by uninstalling and then removing from your `astro.config.mjs` file.
+
+2. Migrate any existing `<Image />` components.
+
+    Change all `import` statements from `@astrojs/image/components` to `astro:assets` to use the new built-in `<Image />` component.
+
+    The supported props are identical with the exception of `aspectRatio`, which is now automatically inferred from the `width` and `height` attributes.
```

Comment:
This is not true:
- `@astrojs/images` supported more properties, notably because its behaviour was different regarding cropping and it needed to support things such as `fitPosition`
- `aspectRatio` was always inferred from `width` and `height` if they existed. Its purpose was to allow you to omit either `width` or `height` as it would use provided-width/height + aspect ratio to calculate the missing dimension.